tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/adapter/ImportApplicationWizardTest.java

```diff
@@ -0,0 +1,181 @@
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftServiceRequirement.RequiredService;
+import org.jboss.tools.openshift.reddeer.utils.DatastoreOS3;
+import org.jboss.tools.openshift.reddeer.utils.OpenShiftLabel;
+import org.jboss.tools.openshift.reddeer.utils.TestUtils;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftExplorerView;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShift3Connection;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShiftProject;
+import org.jboss.tools.openshift.reddeer.view.resources.Service;
+import org.jboss.tools.openshift.reddeer.wizard.importapp.GitCloningWizardPage;
+import org.jboss.tools.openshift.reddeer.wizard.importapp.ImportApplicationWizard;
+import org.jboss.tools.openshift.reddeer.wizard.server.ServerSettingsWizard;
+import org.jboss.tools.openshift.reddeer.wizard.server.ServerSettingsWizardPage;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author jnovak@redhat.com
+ */
+@OCBinary
+@RequiredBasicConnection
+@RequiredProject(
+		name = DatastoreOS3.TEST_PROJECT, 
+		cleanup = true)
+@RequiredService(
+		project = DatastoreOS3.TEST_PROJECT, 
+		service = OpenShiftResources.NODEJS_SERVICE, 
+		template = OpenShiftResources.NODEJS_TEMPLATE)
+@RunWith(RedDeerSuite.class)
+public class ImportApplicationWizardTest {	
+	
+	@InjectRequirement
+	private static OpenShiftProjectRequirement requiredProject;
+	@InjectRequirement
+	private static OpenShiftConnectionRequirement requiredConnection;
+	@InjectRequirement
+	private static OpenShiftServiceRequirement requiredService;
+	private static Service service;
+	
+	@BeforeClass
+	public static void init(){
+		OpenShiftExplorerView openshiftExplorer = new OpenShiftExplorerView();
+		openshiftExplorer.open();
+		
+		OpenShift3Connection connection = openshiftExplorer.getOpenShift3Connection(requiredConnection.getConnection());
+		OpenShiftProject project = connection.getProject(requiredProject.getProjectName());
+		project.expand();
+		service = project.getService(requiredService.getService().getName());
+		
+		assertNotNull("OpenShift service '" + OpenShiftResources.NODEJS_SERVICE 
+					+ "' was not found!", service);
+	}
+	
+	@Before
+	public void cleanUp(){
+		cleanClonnedProjects();
+	}
+	
+	@Test
+	public void testImportOpenShift3AppViaOpenshiftView() {
+		ImportApplicationWizard importWizard = 
+				new ImportApplicationWizard().openFromOpenshiftView(service);
+		
+		importWizard.finish();
+		assertProjectExistsInProjectView(OpenShiftResources.NODEJS_GIT_NAME);
+	}
+	
+	@Test
+	public void testImportOpenShift3AppViaServerAdapterSettings() {
+		ServerSettingsWizard serverWizard = new ServerSettingsWizard().openFromOpenshiftView(service);
+		
+		ImportApplicationWizard importWizard = new ServerSettingsWizardPage().importProject();
+		importWizard.next();
+		
+		importWizard.finish();
```

Comment:
I guess that in this phase, the freshly created project should be pre-selected in "Eclipse Project:" textbox. Please add check for this.